src/libraries/System.Collections/src/System/Collections/BitArray.cs

```diff
@@ -837,7 +837,7 @@ public unsafe void CopyTo(Array array, int index)
                 Vector128<byte> lowerShuffleMask_CopyToBoolArray = Vector128.Create(0, 0x01010101_01010101).AsByte();
                 Vector128<byte> upperShuffleMask_CopyToBoolArray = Vector128.Create(0x02020202_02020202, 0x03030303_03030303).AsByte();
 
-                if (Avx512F.IsSupported && (uint)m_length >= Vector512<byte>.Count)
+                if (Avx512BW.IsSupported && (uint)m_length >= Vector512<byte>.Count)
```

Comment:
Since you're touching this anyways, would you want to update this to use the `xplat` APIs instead?

That is, swap these out:
* `Avx512BW.IsSupported` -> `Vector512.IsHardwareAccelerated`
* `Avx512BW.Shuffle(x, y)` -> `Vector512.Shuffle(x, y)`
* `Avx512F.And(x, y)` -> `x & y`
* `Avx512BW.Min(x, y)` -> `Vector512.Min(x, y)`
* `Avx512F.Store(x, y)` -> `y.Store(x)`

Similar changes could also be made to the `Avx2` path (using `Vector256`) and so on.

--------------

This change to `Avx512BW` is notably "more correct" than `Avx512F`, but notably it's not technically broken today since we require AVX512F+BW+CD+DQ+VL to be provided for any AVX512 support to exist. It's still goodness to fix, but moving away from the platform specific APIs altogether would be even better.

Reply:
About `Vector512.Shuffle(x, y)`
https://github.com/dotnet/runtime/blob/1a51788af474a14cd8708a9a958fce93a47837e6/src/libraries/System.Collections/src/System/Collections/BitArray.cs#L928-L930
It seems that Arm64 might not have the proper instruction to support `VectorXXX.Shuffle<byte>(x, y)`. Could it happen that `VectorXXX.IsHardwareAccelerated` is true but `VectorXXX.Shuffle<byte>(x, y)` has poor performance on some platforms?

Reply:
> It seems that Arm64 might not have the proper instruction to support VectorXXX.Shuffle<byte>(x, y). C

That comment is very outdated, `VectorXXX.Shuffle` covers the usage and does the correct thing.

> Could it happen that VectorXXX.IsHardwareAccelerated is true but VectorXXX.Shuffle<byte>(x, y) has poor performance on some platforms?

Not for the way we're using it here. Theoretically it could happen if you were on a machine from 2005 and didn't have SSSE3 but hit the Vector128.IsHardwareAccelerated code path. This isn't really a concern for anyone using the current versions of .NET and could be adjusted later if it was a concern.

Reply:
Today, `Vector256.IsHardwareAccelerated` strictly implies AVX2 support, and `Vector512.IsHardwareAccelerated` strictly implies AVX-512F+CD+DQ+BW+VL.  Arm64 support for larger vectors comes in the form of Scalable Vector Extensions (SVE), which does not allow acceleration of the fixed-width vector types.

The `Vector128` code in `BitArray` is outdated and could also use the xplat intrinsics.

Reply:
Actually, after the runtime has proper AVX512 masking support https://github.com/dotnet/runtime/issues/87097, the AVX512 path should probably use `kmovq k, m64` and use that mask. This gives a reason to promote https://github.com/dotnet/runtime/issues/87097. Will `xplat Vector512` offer the same capacity?

Reply:
> `Avx512BW.Shuffle(x, y)` -> `Vector512.Shuffle(x, y)`

Today with .NET 10.0.100-preview.3.25201.16, `Vector512.Shuffle(x, y)` generates `vpermb`, but `Avx512BW.Shuffle(x, y)` generates `vpshufb`. Not only is `vpermb` slower than `vpshufb` (https://uops.info/table.html), `vpermb` is actually `AVX512_VBMI`, not just `AVX-512F+CD+DQ+BW+VL`. One might hope `Vector512.Shuffle(x, y)` generates `vpshufb` when possible but this is not done today. It's probably not a good idea to change `Avx512BW.Shuffle(x, y)` to `Vector512.Shuffle(x, y)`.

Reply:
`Shuffle` attempts to normalize behavior across platforms, including zeroing out of range indices and allowing full vector (cross-lane) shuffle/permute. If you don't need that behavior, you can use the new `Vector512.ShuffleNative`, which will emit the expected `vpshufb`. 

> vpermb is actually AVX512_VBMI, not just AVX-512F+CD+DQ+BW+VL

This is not a problem. The baseline ISA set is required for any acceleration of 'Vector512`, but JIT can opportunistically use any ISA supported by the hardware.

Reply:
It looks with the latest .NET 10.0.100-preview.5.25224.4, `Vector512.Shuffle(x, y)` may optimize to `vpshufb`, so no longer a problem.

Reply:
> Since you're touching this anyways, would you want to update this to use the xplat APIs instead?

Should this PR be merged or closed in the interim?

Reply:
Logged https://github.com/dotnet/runtime/issues/116079 to track the move to xplat APIs